### PR TITLE
run on windows, macosx and ubuntu

### PR DESCRIPTION
**Pull request by jswhit**

### CODE REVIEW

File: .github/workflows/miniconda.yml [hunk 1]
```diff
@@ -24,8 +28,9 @@ jobs:
     - name: Python ${{ matrix.python-version }}
       shell: bash -l {0}
       run: |
-        conda create --name TEST python=${{ matrix.python-version }} "numpy=1.17" cython pip pytest hdf5 libnetcdf cftime --strict-channel-priority
+        conda create --name TEST python=${{ matrix.python-version }} numpy cython pip pytest hdf5 libnetcdf cftime --strict-channel-priority
         source activate TEST
+        export PATH="${CONDA_PREFIX}/bin:${CONDA_PREFIX}/Library/bin:$PATH" # so setup.py finds nc-config
```
Comment (ocefpaf) [inline .github/workflows/miniconda.yml:33]: Ah! The GHA was not adding this to the path. This was driving me nuts!

File: .github/workflows/build.yml [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+    - name: Install python dependencies via pip
+      run: |
+        python -m pip install numpy
+        python -m pip install cython
+        python -m pip install cftime
+        python -m pip install pytest
+        python -m pip install check-manifest
+        python -m pip install twine
+        python -m pip install wheel
+        python -m pip install mpi4py
```
Comment (ocefpaf) [inline .github/workflows/build.yml:69]: I don't understand the new pip solver but maybe installing all of these in a single go can be faster than making it solve the current env for a future change in every new installation.

**Review by ocefpaf — APPROVED** [review]
Awesome! Sorry I dropped the ball on #1057 but I'm glad it was a little helpful to get this one done.

File: .github/workflows/build.yml (hunk 2 above)
Comment (jswhit) [reply .github/workflows/build.yml]: Done

**Comment by jswhit** [comment]
Thanks @ocefpaf  - most of this is just copied from PR #1057, so I wouldn't say you dropped the ball. I just carried it the last few meters.

**Pull request merged**